sdk/testing/stepwise/environments/docker/environment.go

```diff
@@ -0,0 +1,851 @@
+			"disable_hostname": true,
+		},
+		"storage": map[string]interface{}{
+			"raft": map[string]interface{}{
+				"path":    "/vault/file",
+				"node_id": n.NodeID,
+			},
+		},
+		"cluster_name":         netName,
+		"log_level":            "TRACE",
+		"raw_storage_endpoint": true,
+		"plugin_directory":     "/vault/config",
+		// disable_mlock is required for working in the Docker environment with
+		// custom plugins
+		"disable_mlock": true,
+		// These are being provided by docker-entrypoint now, since we don't know
+		// the address before the container starts.
+		//"api_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port),
+		//"cluster_addr": fmt.Sprintf("https://%s:%d", n.Address.IP, n.Address.Port+1),
+	}
+	cfgJSON, err := json.Marshal(vaultCfg)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(n.WorkDir, "local.json"), cfgJSON, 0644)
+	if err != nil {
+		return err
+	}
+	// setup plugin bin copy if needed
+	copyFromTo := map[string]string{
+		n.WorkDir: "/vault/config",
+		caDir:     "/usr/local/share/ca-certificates/",
+	}
+	if pluginBinPath != "" {
+		base := path.Base(pluginBinPath)
+		copyFromTo[pluginBinPath] = filepath.Join("/vault/config", base)
+	}
+
+	r := &Runner{
+		dockerAPI: cli,
+		ContainerConfig: &container.Config{
+			Image: "vault",
+			Entrypoint: []string{"/bin/sh", "-c", "update-ca-certificates && " +
+				"exec /usr/local/bin/docker-entrypoint.sh vault server -log-level=trace -dev-plugin-dir=./vault/config -config /vault/config/local.json"},
+			Env: []string{
+				"VAULT_CLUSTER_INTERFACE=eth0",
+				"VAULT_API_ADDR=https://127.0.0.1:8200",
+				fmt.Sprintf("VAULT_REDIRECT_ADDR=https://%s:8200", n.Name()),
+			},
+			Labels:       nil,
+			ExposedPorts: nat.PortSet{"8200/tcp": {}, "8201/tcp": {}},
+		},
+		ContainerName: n.Name(),
+		NetName:       netName,
+		CopyFromTo:    copyFromTo,
+	}
+
+	n.container, err = r.Start(context.Background())
+	if err != nil {
+		return err
+	}
+
+	n.Address = &net.TCPAddr{
+		IP:   net.ParseIP(n.container.NetworkSettings.IPAddress),
+		Port: 8200,
+	}
+	ports := n.container.NetworkSettings.NetworkSettingsBase.Ports[nat.Port("8200/tcp")]
+	if len(ports) == 0 {
+		n.Cleanup()
+		return fmt.Errorf("could not find port binding for 8200/tcp")
+	}
+	n.HostPort = ports[0].HostPort
+
+	return nil
+}
+
+// DockerClusterOptions has options for setting up the docker cluster
+type DockerClusterOptions struct {
+	KeepStandbysSealed bool
+	RequireClientAuth  bool
+	SkipInit           bool
+	CACert             []byte
+	NumCores           int
+	tmpDir             string
+	PluginTestBin      string
+	// SetupFunc is called after the cluster is started.
+	SetupFunc func(t testing.T, c *DockerCluster)
+	CAKey     *ecdsa.PrivateKey
+}
+
+//
+// test methods/functions
+//
+
+// testWaitHealthMatches checks health
+func testWaitHealthMatches(ctx context.Context, client *api.Client, ready func(response *api.HealthResponse) error) error {
```

Comment:
The naming makes these functions seem like they are used in test files, but they are more of helpers in the implementation to check on state. Maybe change `test*` to check/ensure/verify, etc.

Reply:
Agree, I updated the names from `testWait*` to `ensure*` in 61e1d39ff